### PR TITLE
Adds Telecrystal Fuel Rods

### PR DESCRIPTION
**Pull request by TheFakeElon**
## About The Pull Request

Adds a "telecrystal fuel rod" to the traitor uplink. up to eight telecrystals can be added into one rod. Once a rod is fully depleted in a reactor, you can harvest your newly transmutated crystals. However, controlling the reactor becomes extremely difficult when using these due to their high energy output

## Why It's Good For The Game

Makes the reactor useful to tots, Gives a reason to make lots of highly autistic cooling setups that would be needed to run this reactor at a decent production rate.

## Changelog
:cl:
add: Adds the telecrystal fuel rod.
/:cl:

### CODE REVIEW

**Comment by ZombiesVsPlants** [comment]
Love it

**Comment by Ohagi-Chan** [comment]
What's the reward for fully depleting the rod?

**Comment by Ohagi-Chan** [comment]
Free TC?

**Comment by TheFakeElon** [comment]
yes

**Comment by TheFakeElon** [comment]
I might make this harder to control after seeing jojo meme the reactor

**Comment by thelaughingbomb** [comment]
look, i dont know HOW i did it, it just happens

**Comment by TheFakeElon** [comment]
_said the plasmeme with 400tc_

**Comment by TheFakeElon** [comment]
done! :D

File: austation/code/modules/power/fuel_rods.dm [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		var/profit = round(telecrystal_amount * multiplier, 1)
+		to_chat(user, "<span class='notice'>You remove [profit] telecrystals from the [src].</span>")
+		var/obj/item/stack/telecrystal/tc = new(get_turf(src))
+		tc.amount = profit
+		expended = TRUE
+		telecrystal_amount = 0
+		return
+
+	else
+		to_chat(user, "<span class='warning'>The [src] has not fissiled enough to fully grow the sample. the progress bar shows it is [min(depletion / 40 * 100, 100)]% complete. </span>")
```
Comment (MCterra10) [inline austation/code/modules/power/fuel_rods.dm:66]:
```suggestion
		to_chat(user, "<span class='warning'>The [src] has not fissiled enough to fully grow the sample. The progress bar shows it is [min(depletion / 40 * 100, 100)]% complete.</span>")
```
grammar

File: austation/code/modules/power/fuel_rods.dm [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+/obj/item/twohanded/required/fuel_rod/telecrystal/examine(mob/user)
+	. = ..()
+	if(expended)
+		. += "<span class='warning'>The material slots have been removed.</span>"
```
Comment (MCterra10) [inline austation/code/modules/power/fuel_rods.dm:71]: doesn't make sense. maybe talk about the uranium-dioxide being exhausted in the slots?

**Review by MCterra10 — CHANGES_REQUESTED** [review]
also make sure that when you proceed a variable name with "the" to use "\the" or "\The" (depending on capitalization). This is a macro to make sure your code handles proper nouns correctly and is generally good practice.

File: austation/code/modules/power/fuel_rods.dm [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+/obj/item/twohanded/required/fuel_rod/telecrystal/attack_self(mob/user)
+	if(expended)
+		to_chat(user, "<span class='notice'>You have already removed the telecrystals from the [src].</span>")
+		return
+
+	if(grown)
+		var/profit = round(telecrystal_amount * multiplier, 1)
+		to_chat(user, "<span class='notice'>You remove [profit] telecrystals from the [src].</span>")
+		var/obj/item/stack/telecrystal/tc = new(get_turf(src))
```
[the remark this answers is not in the document]
Comment (TheFakeElon) [reply austation/code/modules/power/fuel_rods.dm:59]: fuel rods require two hands to use, there is no inactive hand :)

File: austation/code/modules/power/fuel_rods.dm (hunk 2 above)
Comment (TheFakeElon) [reply austation/code/modules/power/fuel_rods.dm:71]: ehhh, the plutonium is more radioactive so I don't see why it wouldn't work. I'll make it say something about the slots being slagged due to the high heat

**Review by MCterra10 — APPROVED** [review]

**Pull request merged**